[Bug][fully_async] DetachActorWorker crashes on FSDP1 when recomputing old_log_prob

### Summary

Fully Async training crashes when all of the following conditions are met:

* The actor uses FSDP1.
* `algorithm.rollout_correction.bypass_mode=False`.
* `FullyAsyncTrainer` recomputes `old_log_prob`.

No source-code modification is required to reproduce the failure.

`DetachActorWorker` currently treats FSDP1 and FSDP2 as sharing the same temporary model-state save/restore handlers for FSDP1 and FSDP2. As a result, an FSDP1 model is passed to `fsdp2_sharded_save_to_cpu()`, which requires DTensor parameters and raises:

```text
AssertionError: No DTensor-type parameters found in the model.
FSDP2 sharding may not be enabled.
```

The default Fully Async configuration uses `bypass_mode=True`, which reuses rollout log-probabilities and does not need to temporarily preserve the policy version for `old_log_prob` recomputation. Therefore, the affected save/restore path is not exercised.

### System Info

* Branch: `main`
* verl version: `0.9.0.dev0`
* Commit: `983cb0f24443f87b3d161fad318445130a620b07`

### Reproduction

Start from the Fully Async example script and add the following overrides:

```bash
actor_rollout_ref.actor.fsdp_config.strategy=fsdp \
algorithm.rollout_correction.bypass_mode=False
```

Both overrides are required to enter the failing path:

* `actor_rollout_ref.actor.fsdp_config.strategy=fsdp` creates an FSDP1 actor.
* `algorithm.rollout_correction.bypass_mode=False` enables recomputation of `old_log_prob`.

The resulting execution path is:

```text
FullyAsyncTrainer._fit_compute_log_prob
  -> FullyAsyncTrainer._compute_old_log_prob
  -> DetachActorWorker.save_model_to_cpu
  -> fsdp2_sharded_save_to_cpu
```

The failure occurs when `_compute_old_log_prob()` temporarily saves the current actor state before switching policy versions for `old_log_prob` recomputation:

```python
if self.local_trigger_step == 1:
    self.actor_rollout_wg.save_model_to_cpu(1)
```

### Observed behavior

```text
Traceback (most recent call last):
  File ".../fully_async_trainer.py", line 680, in _compute_old_log_prob
    self.actor_rollout_wg.save_model_to_cpu(1)
  File ".../experimental/separation/engine_workers.py", line 127, in save_model_to_cpu
    self.cpu_saved_models[n] = self.copy_handler(
        self.actor.engine.module
    )
  File ".../utils/fsdp_utils.py", line 1131, in fsdp2_sharded_save_to_cpu
    assert global_spec is not None, (
        "No DTensor-type parameters found in the model. "
        "FSDP2 sharding may not be enabled."
    )
AssertionError: No DTensor-type parameters found in the model.
FSDP2 sharding may not be enabled.
```

### Root cause

`DetachActorWorker._get_strategy_handlers()` groups FSDP1, FSDP2, and VeOmni under the same handlers:

```python
if strategy in ["fsdp", "fsdp2", "veomni"]:
    from verl.utils.fsdp_utils import (
        fsdp2_sharded_load_from_cpu,
        fsdp2_sharded_save_to_cpu,
    )

    self._strategy_handlers = (
        fsdp2_sharded_save_to_cpu,
        fsdp2_sharded_load_from_cpu,
    )
```

Therefore, when the resolved actor strategy is `fsdp`, `save_model_to_cpu()` invokes:

```python
fsdp2_sharded_save_to_cpu(self.actor.engine.module)
```

However, `fsdp2_sharded_save_to_cpu()` is specifically implemented for FSDP2 models whose sharded parameters are DTensors:

```python
for param_name, param in model.named_parameters():
    if not isinstance(param, DTensor):
        cpu_tensor = param.detach().cpu()
        cpu_sharded_state[param_name] = (cpu_tensor, None)
        continue

    if global_spec is None:
        global_spec = param._spec

assert global_spec is not None, (
    "No DTensor-type parameters found in the model. "
    "FSDP2 sharding may not be enabled."
)
```

FSDP1 uses FSDP-managed parameters rather than FSDP2 DTensor parameters. Consequently, no parameter initializes `global_spec`, and the FSDP2-specific assertion is raised.

The restore path has the same coupling: `"fsdp"` is assumed to have produced the FSDP2 `(cpu_sharded_state, global_spec)` saved-state representation.

### Expected behavior

Fully Async with FSDP1 and `algorithm.rollout_correction.bypass_mode=False` should not fail inside an FSDP2-only helper.

One of the following behaviors would be expected:

1. **Support temporary policy-state preservation for FSDP1**

   Route FSDP1 actors to an FSDP1-compatible state save/restore implementation, while keeping `fsdp2_sharded_save_to_cpu()` and `fsdp2_sharded_load_from_cpu()` limited to strategies using the expected
   FSDP2 DTensor representation.

2. **Reject the configuration early**

   If temporary policy-state save/restore for FSDP1 is intentionally unsupported, configuration validation should reject the combination of FSDP1 and `bypass_mode=False` with a clear error explaining that `old_log_prob` recomputation is unsupported for FSDP1.

--- a/verl/experimental/separation/engine_workers.py
+++ b/verl/experimental/separation/engine_workers.py
@@ -72,15 +72,25 @@ class DetachActorWorker(ActorRolloutRefWorker):
 
         strategy = self.config.actor.strategy
 
-        # NOTE: VeOmni internally uses FSDP2 for data parallelism (VeOmniEngine inherits from
+        # NOTE: FSDP1 shards parameters as plain (flat) tensors, while FSDP2 shards them as
+        # DTensors, so the two need different save/restore utilities.
+        #
+        # VeOmni internally uses FSDP2 for data parallelism (VeOmniEngine inherits from
         # FSDPEngine and sets data_parallel_mode="fsdp2"), so its model parameters are DTensors
         # that are compatible with FSDP2's sharded save/load utilities.
         #
-        # CAVEAT: When VeOmni's param_offload=True, parameters may reside on CPU at the time of
-        # save/restore. The current fsdp2_sharded_save_to_cpu / fsdp2_sharded_load_from_cpu
-        # assume parameters are on GPU. Callers should ensure the model is loaded back to GPU
-        # before calling save_model_to_cpu / restore_model_from_cpu in offload scenarios.
-        if strategy in ["fsdp", "fsdp2", "veomni"]:
+        # CAVEAT: When param_offload=True, parameters may reside on CPU at the time of
+        # save/restore. The current sharded save/load helpers assume parameters are on GPU.
+        # Callers should ensure the model is loaded back to GPU before calling
+        # save_model_to_cpu / restore_model_from_cpu in offload scenarios.
+        if strategy == "fsdp":
+            from verl.utils.fsdp_utils import (
+                fsdp1_sharded_load_from_cpu,
+                fsdp1_sharded_save_to_cpu,
+            )
+
+            self._strategy_handlers = (fsdp1_sharded_save_to_cpu, fsdp1_sharded_load_from_cpu)
+        elif strategy in ["fsdp2", "veomni"]:
             from verl.utils.fsdp_utils import (
                 fsdp2_sharded_load_from_cpu,
                 fsdp2_sharded_save_to_cpu,
@@ -114,9 +124,9 @@ class DetachActorWorker(ActorRolloutRefWorker):
         """
         Save the current model state to CPU memory.
 
-        For FSDP/FSDP2/VeOmni strategies, this uses fsdp2_sharded_save_to_cpu which
-        expects model parameters to be on GPU (as DTensors). If VeOmni param_offload
-        is enabled, ensure the model has been reloaded to GPU before calling this method.
+        For FSDP/FSDP2/VeOmni strategies, this uses the strategy's sharded save helper, which
+        expects model parameters to be on GPU. If param_offload is enabled, ensure the model
+        has been reloaded to GPU before calling this method.
 
         Args:
             n: Identifier/Key for the saved model state.
@@ -131,9 +141,9 @@ class DetachActorWorker(ActorRolloutRefWorker):
         """
         Restore the model state from CPU memory.
 
-        For FSDP/FSDP2/VeOmni strategies, the saved state is a tuple of
+        For FSDP2/VeOmni strategies, the saved state is a tuple of
         (cpu_sharded_state, global_spec) produced by fsdp2_sharded_save_to_cpu.
-        For Megatron, the saved state is passed directly to the restore handler.
+        For FSDP1 and Megatron, the saved state is passed directly to the restore handler.
 
         Args:
             n: Identifier/Key for the saved model state to restore.
@@ -141,7 +151,7 @@ class DetachActorWorker(ActorRolloutRefWorker):
         if n in self.cpu_saved_models:
             strategy = self.config.actor.strategy
 
-            if strategy in ["fsdp", "fsdp2", "veomni"]:
+            if strategy in ["fsdp2", "veomni"]:
                 cpu_sharded_state, global_spec = self.cpu_saved_models[n]
                 self.restore_handler(self.actor.engine.module, cpu_sharded_state, global_spec)
             else:

--- a/verl/utils/fsdp_utils.py
+++ b/verl/utils/fsdp_utils.py
@@ -1090,6 +1090,47 @@ def merged_lora_context(actor, backup_adapters=False):
             fsdp_merge_unmerge(actor, do_merge=False)
 
 
+def fsdp1_sharded_save_to_cpu(model: torch.nn.Module) -> dict[str, torch.Tensor]:
+    """
+    Sharded Save for FSDP1: each rank copies its own local parameter shards to CPU memory.
+
+    FSDP1 exposes flat, already-sharded ``torch.nn.Parameter`` tensors (or views into them when
+    ``use_orig_params=True``) instead of DTensors, so the sharding layout is implicit in the local
+    shapes and does not need to be recorded.
+
+    Args:
+        model: FSDP1-wrapped model.
+
+    Returns:
+        Dictionary mapping parameter name to its local shard on CPU.
+    """
+    cpu_sharded_state = {}
+    for param_name, param in model.named_parameters():
+        cpu_sharded_state[param_name] = param.data.detach().to("cpu", copy=True)
+    return cpu_sharded_state
+
+
+def fsdp1_sharded_load_from_cpu(model: torch.nn.Module, cpu_sharded_state: dict[str, torch.Tensor]) -> None:
+    """
+    Sharded Load for FSDP1: each rank copies its own CPU shards back into the live parameters.
+
+    The model must be in the same sharding state as when :func:`fsdp1_sharded_save_to_cpu` was
+    called, so that local shapes still match.
+
+    Args:
+        model: FSDP1-wrapped model to be restored.
+        cpu_sharded_state: Shards saved by :func:`fsdp1_sharded_save_to_cpu` on this rank.
+    """
+    with torch.no_grad():
+        for param_name, param in model.named_parameters():
+            # Skip parameters not in the saved state (e.g., newly added parameters)
+            if param_name not in cpu_sharded_state:
+                continue
+            param.data.copy_(cpu_sharded_state[param_name].to(param.device))
+
+    dist.barrier()
+
+
 def fsdp2_sharded_save_to_cpu(
     model: torch.nn.Module,
 ) -> tuple[dict[str, tuple[torch.Tensor, DTensorSpec]], DTensorSpec]:

